ForkMe-Mobile/app/src/main/java/dlei/forkme/gui/activities/BaseActivity.java

```diff
@@ -0,0 +1,204 @@
+                        mFindPeople,
+                        new SectionDrawerItem().withName(R.string.drawer_settings),
+                        mSettings,
+                        new SectionDrawerItem().withName(R.string.drawer_about),
+                        mAbout
+                )
+                .withOnDrawerItemClickListener(new Drawer.OnDrawerItemClickListener() {
+                    @Override
+                    public boolean onItemClick(View view, int position, IDrawerItem drawerItem) {
+                        if (drawerItem == mStars) {
+                            Intent intent = new Intent(getApplicationContext(), UserStarsActivity.class);
+                            intent.putExtra("userLogin", AppSettings.sUserLogin);
+                            intent.putExtra("userName", AppSettings.sUserName);
+                            intent.putExtra("userAvatarUrl", AppSettings.sUserAvatarUrl);
+                            startActivity(intent);
+                        } else if (drawerItem == mTrending) {
+                            Intent intent = new Intent(getApplicationContext(), TrendingRepositoriesActivity.class);
+                            startActivity(intent);
+
+                        } else if (drawerItem == mFindPeople) {
+                            Intent intent = new Intent(getApplicationContext(), MergeMeActivity.class);
+                            startActivity(intent);
+
+                        } else if (drawerItem == mYourRepositories) {
+                            Intent intent = new Intent(getApplicationContext(), UserRepositoriesViewActivity.class);
+                            intent.putExtra("userLogin", AppSettings.sUserLogin);
+                            startActivity(intent);
+
+                        } else if (drawerItem == mSettings) {
+                            Intent intent = new Intent(getApplicationContext(), SettingsActivity.class);
+                            startActivity(intent);
+
+                        } else if (drawerItem == mAbout) {
+                            Intent intent = new Intent(getApplicationContext(), AboutActivity.class);
+                            startActivity(intent);
+                        }
+                        Log.d("Menu clicked: ", "position: " + position + " item: " + drawerItem);
+                        return true;
+                    }
+                })
+
+                .withSavedInstance(savedInstanceState)
+                .build();
+        Log.d("BaseActivity: ", "inflateNavDrawer: " + mNavDrawer);
+    }
+
+
+    public void inflateNavDrawer(Bundle savedInstanceState, String currentActivity) {
+        Log.d("BaseActivity: ", "inflateNavDrawer: called, currentActivity: " + currentActivity);
+        this.inflateNavDrawer(savedInstanceState);
+
+        // Disable clicking on current activity.
+        if (currentActivity.equals(UserStarsActivity.class.getSimpleName())) {
```

Comment:
Not sure what's going on here. But there should be a much better way to do what you're trying to achieve without checking the class name.